### PR TITLE
Delete report karma award message

### PR DESCRIPTION
**Pull request by Desiders**
Closes https://github.com/bomzheg/KarmaBot/issues/129

### CODE REVIEW

**Comment by Desiders** [comment]
Should I recreate or you resolve conflict?

**Comment by Desiders** [comment]
I tried to resolve conflicts, but in my branch I use `message.` instead of `callback_query.message.`. I think it will not complicate to merge branches.

File: app/config/main.py [hunk 1]
```diff
@@ -40,5 +40,6 @@ def load_config(config_dir: Path = None) -> Config:
         tg_client=TgClientConfig(bot_token=_bot_token),
         storage=load_storage(config_file_data["storage"]),
         report_karma_award=config_file_data.get("report_karma_award", 0),
+        time_to_remove_report_karma_award_messages=config_file_data.get("time_to_remove_report_karma_award_messages", 300),
```
Comment (Lamroy95) [inline app/config/main.py:43]: Let's call it `report_award_cleanup_delay` and set default 1 hour:
```suggestion
        report_award_cleanup_delay=config_file_data.get("report_award_cleanup_delay", 3600),
```

File: app/handlers/moderator.py [hunk 2]
```diff
@@ -341,21 +343,28 @@ async def approve_report_handler(
             reward_amount=config.report_karma_award,
             bot=bot,
         )
-        await callback_query.message.edit_text(
+        await message.edit_text(
             "<b>{reporter}</b> получил <b>+{reward_amount}</b> кармы "
-            "в награду за репорт{admin_url}".format(
+            "в награду за репорт!{admin_url}".format(
```
Comment (Lamroy95) [inline app/handlers/moderator.py:348]: Is this "!" left from rebase?  
It is removed in target branch

File: app/handlers/moderator.py [hunk 3]
```diff
@@ -341,21 +343,28 @@ async def approve_report_handler(
...
                 reporter=hd.quote(karma_change_result.karma_event.user_to.fullname),
                 reward_amount=config.report_karma_award,
                 admin_url=hidden_link(user.link),
             )
         )
+        if config.time_to_remove_report_karma_award_messages > 0:
```
Comment (Lamroy95) [inline app/handlers/moderator.py:354]:
```suggestion
        if config.report_award_cleanup_delay > 0:
```

File: app/handlers/moderator.py [hunk 4]
```diff
@@ -341,21 +343,28 @@ async def approve_report_handler(
...
+            asyncio.create_task(
+                delete_message(
+                    message=message,
+                    sleep_time=config.time_to_remove_report_karma_award_messages,
```
Comment (Lamroy95) [inline app/handlers/moderator.py:358]:
```suggestion
                    sleep_time=config.report_award_cleanup_delay,
```

File: app/models/config/__init__.py [hunk 5]
```diff
@@ -26,6 +26,7 @@ class Config:
     time_to_cancel_actions: int = 60
     time_to_remove_temp_messages: int = 30
     report_karma_award: int = 0
+    time_to_remove_report_karma_award_messages: int = 300  # If time in seconds less than 1, then messages will not be deleted
```
Comment (Lamroy95) [inline app/models/config/__init__.py:29]:
```suggestion
    report_award_cleanup_delay: int = 3600  # If time in seconds less than 1, then messages will not be deleted
```

**Review by Lamroy95 — CHANGES_REQUESTED** [review]

File: app/models/config/__init__.py (hunk 5 above)
Comment (Desiders) [reply app/models/config/__init__.py:29]: @bomzheg, what do you think about this suggestion?

There was also a suggestion to use `reward_message_ttl`.

**Comment by Lamroy95** [comment]
What about conflicts?

**Review by Lamroy95 — CHANGES_REQUESTED** [review]
Please fix merge/rebase

**Review by Lamroy95 — APPROVED** [review]

**Review by Lamroy95 — APPROVED** [review]

**Review by bomzheg — APPROVED** [review]

**Pull request merged**